packages/katello/katello/hostname-change.rb

```diff
@@ -239,6 +292,107 @@ def setup_opt_parser
       @opt_parser.parse!
     end
 
+    def dns_managed?
+      @scenario_answers['foreman_proxy']['dns'] &&
+        @scenario_answers['foreman_proxy']['dns_managed'] &&
+        @scenario_answers['foreman_proxy']['dns_provider'] == 'nsupdate'
+    end
+
+    def should_update_dns?
+      dns_managed? && !@options[:skip_dns]
+    end
+
+    def answers_dns_values
+      new_vals = OpenStruct.new
```

Comment:
I think the recent refactoring makes the need for the OpenStruct a lot less compelling. what do you think?

Just using a hash will make your pre-check code simpler as well - you can just do this:

```
if @answers_dns_values.values.any? { |v| v.nil? }
  # show the error
end

```

Reply:
@jturel For the error message I needed the key also (`field`), so the resulting code isn't that much cleaner here.  But I still think it's better in case we need to change any of the keys in the future, since they won't be hard-coded.